### PR TITLE
Update "What's new" table

### PR DESCRIPTION
**Pull request by bethcollins92**
## Done

Fix inconsistencies in what's new table

Fixes [list issues/bugs if needed]

## QA

- Open [demo](https://vanilla-framework-4441.demos.haus/)
- Check what's new table and side nav are consistent with correct labels

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical-web-and-design/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [x] Documentation side navigation should be updated with the relevant labels.

## Screenshots

[if relevant, include a screenshot or screen capture]

### CODE REVIEW

**Comment by bartaz** [comment]
The old `p-tab-button` component name and class name should be mentioned in "What's new" as deprecated

**Comment by bartaz** [comment]
@bethcollins92 Please add the deprecated class names to the list as well, this is very helpful later when we remove deprecated stuff in next major version.

**Comment by bethcollins92** [comment]
@bartaz Added the deprecated classes to the table

File: templates/docs/whats-new.md [hunk 1]
```diff
@@ -20,23 +20,46 @@ When we add, make significant updates, or deprecate a component we update their
     </tr>
   </thead>
   <tbody>
-    <!-- 3.3.1 -->
+  <!-- 3.3.0 -->
+    <tr>
+      <th><a href="/docs/patterns/navigation">Side navigation</a></th>
```
Comment (bartaz) [inline templates/docs/whats-new.md:25]:
```suggestion
      <th><a href="/docs/patterns/navigation#side-navigation">Side navigation</a></th>
```

File: templates/docs/whats-new.md [hunk 2]
```diff
@@ -20,23 +20,46 @@ When we add, make significant updates, or deprecate a component we update their
...
+      <td>
+        <span class="p-status-label--information">Updated</span>
+      </td>
+      <td>3.3.0</td>
+      <td>We've made some changes to the side navigation, improving accessibility and renaming some classes.</td>
+    </tr>
+    <tr>
+      <th><a href="/docs/patterns/navigation">Side navigation</a></th>
```
Comment (bartaz) [inline templates/docs/whats-new.md:33]:
```suggestion
      <th><a href="/docs/patterns/navigation#side-navigation">Side navigation</a></th>
```

**Review by bartaz — APPROVED** [review]
LGTM, thanks

**Pull request merged**